### PR TITLE
Fix for #26284 - Incorrect Tax showing for online Credit Memo

### LINKED ISSUE #26284: Incorrect Tax showing for online Credit Memo
### Preconditions (*)

1. latest Magento version
2. online payment method require like Braintree or PayPal
3. create an order with at least two items
4. place an order and create invoice separately
5. create a credit memo from one of the invoice ( make sure online credit memo)

### Steps to reproduce (*)

1. place an order with two products and online payment method. each product has tax. 
https://www.screencast.com/t/POREIL81nvOZ
2. create two invoices for individual products.
3. now click create credit memo from one of invoice.

### Expected result (*)

1.  credit memo should include tax of that particular item only.

### Actual result (*)

1. invoice showing tax correct but credit memo calculating tax of whole order.
https://www.screencast.com/t/i8cIiuWvt
https://www.screencast.com/t/WLMGPPvtzTy

### PR DESCRIPTION
**Pull request by nikoelgatito**
### Description (*)
Fix for Issue: Incorrect Tax showing for online Credit Memo #26284
Link: Fixes magento/magento2#26284

The issue was caused by the fact that the $allowedTax and $allowedBaseTax variables were set to the same values without taking into consideration if the current creditmemo is being created for an invoice. The commit makes sure to subtract the refunded tax for any credit memo that is associated with the current invoice.

### Manual testing scenarios (*)

1. Place an order with at least 2 products.
2. Create 2 different invoices, each covering a single product.
3. Create a credit memo for one of the invoice.

**Result:** The credit memo tax total should only reflect the non-refunded invoice tax total instead of the tax for the entire order.

### Questions or comments
Ideally a getCreditmemosCollection(), getTaxRefunded() and getBaseTaxRefunded() functions could also be integrated directly into the Magento\Sales\Model\Order\Invoice class based on the submitted changes.

### Contribution checklist (*)
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Sales/Model/Order/Creditmemo/Total/Tax.php [hunk 1]
```diff
@@ -95,6 +95,23 @@ public function collect(\Magento\Sales\Model\Order\Creditmemo $creditmemo)
...
+                $invoiceCreditmemos = $order->getCreditmemosCollection()->addFieldToFilter('invoice_id', $invoice->getId());
+                
+                foreach ($invoiceCreditmemos as $invoiceCreditmemo) {
+                    $taxRefunded += $invoiceCreditmemo->getTaxAmount();
+                    $baseTaxRefunded += $invoiceCreditmemo->getBaseTaxAmount();
+                }
+            }
+            
+            $allowedTax = $invoice->getTaxAmount() - $taxRefunded - $creditmemo->getTaxAmount();
+            $allowedBaseTax = $invoice->getBaseTaxInvoiced() - $baseTaxRefunded - $creditmemo->getBaseTaxAmount();
```
Comment (samicoman) [inline app/code/Magento/Sales/Model/Order/Creditmemo/Total/Tax.php]: @nikoelgatito it should be `$invoice->getBaseTaxAmount()`

Comment (nikoelgatito) [reply app/code/Magento/Sales/Model/Order/Creditmemo/Total/Tax.php]: @samicoman Thank you for pointing this error. I have submitted a fix through the commit #3ac0d99.
https://github.com/magento/magento2/pull/27301/commits/3ac0d9927f98520743baeed13d01d20021397f95

**Review by nikoelgatito — COMMENTED** [review]
Changes ready for approval and production.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @nikoelgatito. Thanks for collaboration. Can you please fix failed tests and cover this scenario by mftf test?
Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests.

**Comment by engcom-Kilo** [comment]
@nikoelgatito
 Please sign CLA.

**Comment by engcom-Kilo** [comment]
Working with MFTF test.

**Comment by engcom-Kilo** [comment]
Seems this case can not cover with MFTF test.
Because for test need use Online Payment method.

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen if you are going to continue working on it.
Make sure CLA is signed.
Thank you.

**Pull request closed without merging**